docs/src/content/docs/utilities/Injectors/inject-query-params.md

```diff
@@ -45,25 +63,136 @@ If we want to get the value for a specific query param, we can pass the name of
 class TestComponent {
 	searchParam = injectQueryParams('search'); // returns a signal with the value of the search query param
 
-	filteredUsers = computedFrom(
-		[this.searchParam],
-		switchMap((searchQuery) =>
-			this.userService.getUsers(searchQuery).pipe(startWith([])),
-		),
+	filteredUsers = computedAsync(
+		() => this.userService.getUsers(searchQuery ?? ''),
+		{ initialValue: [] },
 	);
 }
 ```
 
-Or, if we want to transform the query params, we can pass a function to `injectQueryParams`.
+If we want to additional transform the value into any shape, we can pass a `transform` function.
 
 ```ts
-@Component()
+@Component({
+	template: `
+		Number: {{ number1() }} New number: {{ newNumber() }}
+	`,
+})
 class TestComponent {
-	queryParamsKeys = injectQueryParams((params) => Object.keys(params)); // returns a signal with the keys of the query params
+	number1 = injectQueryParams('id', { transform: numberAttribute }); // returns a signal with the value of the search query param
```

Comment:
I'd like to have examples which are meaningful. So let's convert this to a pageNumber and use better var names.

```suggestion
	pageNumber = injectQueryParams('pageNumber', { transform: numberAttribute });
```